core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -561,9 +544,10 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       setNewProtocolWithFeaturesEnabledByMetadata(newMetadataTmp)
     }
 
-
-
-    assertNoDeletionVectors(newMetadataTmp, spark)
+    DeletionVectorUtils.assertDeletionVectorsNotReadable(
```

Comment:
THIS is the place where we should be checking that the DV table property is not set. Irrespective of whether DVs are readable. Because if they are not and it's set, that gets us into the `illegal` state I described.

Reply:
Updated the PR. Please take another look at it.